### DISCUSSION TITLE
Uploadable favicon

### DISCUSSION THREAD

**Discussion by hubertmeier** [arendst/Tasmota#19167] [Ideas]
It would be nice to have a customisable favicon for the Web UI, if I store the link on the Homescreen, a proper icon would be nice. Is there a place in the filesystem, at least in the esp32 versions? The icon could be uploaded in the UI ...

Thank you

**Comment by s-hadinger** [comment]
This should be doable on esp32 if someone wants to take the time for it. I'm not sure what value it would bring though

**Reply by sfromis** [reply]
The favicon is in source file `HTTP_HEADER1_ES6.h` if you want to play with it. Beware that this file is needs to be converted to the compressed format if you modify.

**Reply by hubertmeier** [reply]
The favicon is the icon which would be stored on the android Homescreen or in the Favorites of a browser. Especially the Homescreen is of interest. Now I have only a blue rectangle with a number in it... I have different tasmota switches, the icons are all the same.

The idea was to upload an icon to the web UI which is then part of this switch.

I know, this feature is not really important... It's just an idea. Thank you

**Reply by sfromis** [reply]
Looks like the reason you do not get the default Tasmota favicon on the Android home screen is that it is a very small size of 16x16 pixels. IIRC, on earlier versions it would be scaled up to a very blurry mess, but now seems to be rejected as unsuitable. Larger icons are accepted.

**Reply by sfromis** [reply]
BTW. If you want to create customized browser shortcuts on Android, the app "Shortcut Maker" looks useful, allowing you to set icons.
